src/main/kotlin/com/salesforce/dynamodbv2/mt/repo/MtTableDescriptionRepo.kt

```diff
@@ -0,0 +1,67 @@
+/* Copyright (c) 2019, Salesforce.com, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause.
+ * For full license text, see LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause.
+ */
+package com.salesforce.dynamodbv2.mt.repo
+
+import com.amazonaws.services.dynamodbv2.model.CreateTableRequest
+import com.amazonaws.services.dynamodbv2.model.TableDescription
+import com.salesforce.dynamodbv2.mt.mappers.MtAmazonDynamoDb.TenantTable
+
+/**
+ * Interface for managing table metadata of multi-tenant virtual tables within mt-dynamo.
+ *
+ * @author msgroi
+ */
+interface MtTableDescriptionRepo {
+
+    /**
+     * Create a multitenant (virtual) table with specs defined in param createTableRequest under the given
+     * multitenant context's namespace.
+     *
+     * @param createTableRequest specs of table to create
+     * @return the table description of said table
+     */
+    fun createTable(createTableRequest: CreateTableRequest): TableDescription
+
+    /**
+     * @param tableName to look up
+     * @return the table description of the given virtual table under the current multi-tenant context
+     */
+    fun getTableDescription(tableName: String): TableDescription
+
+    /**
+     * Delete the designated virtual table metadata.
+     * @param tableName tpo de.lete
+     * @return the table description of the virtual table deleted
+     */
+    fun deleteTable(tableName: String): TableDescription
+
+    /**
+     * Utility to enumerate all virtual table metadata managed by this instance. Return up to @param limit results,
+     * starting after @param exclusiveStartTableMetadata if specified.
+     *
+     * @return a list of TenantTableMetadata objects, with a lastEvaluatedTable populated to the last table name if the
+     * result set fully populated @param limit results, null otherwise.
+     */
+    fun listVirtualTableMetadatas(
+        limit: Int,
+        exclusiveStartTableMetadata: TenantTableMetadata?
+    ): ListMetadataResult
+
+    /**
+     * TODO: these overloaded listVirtualTableMetadatas are placeholders from implementing this interface in Java.
+     * (see https://stackoverflow.com/questions/47070968/kotlin-default-arguments-in-interface-bug)
+     * Once the impl is converted to Kotlin, all the below signatures can be removed with defaulted params.
+     */
+    fun listVirtualTableMetadatas(exclusiveStartTableMetadata: TenantTableMetadata?): ListMetadataResult
+    fun listVirtualTableMetadatas(limit: Int): ListMetadataResult
+    fun listVirtualTableMetadatas(): ListMetadataResult
+    data class ListMetadataResult(val metadataList: List<TenantTableMetadata>, val lastEvaluatedTable: TenantTableMetadata?)
+
+    data class TenantTableMetadata(val tenantTable: TenantTable, val createTableRequest: CreateTableRequest)
```

Comment:
I wonder if it would be more consistent with other interfaces (like `scan` and `getRecords`) to return a list of `CreateTableRequests` that contain the context (maybe an `MtCreateTableRequest`). This class threw me off at first, because the `tenantTable` field contains the `virtualTableName` which the same as the `tableName` in the `CreateTableRequest`, correct? IIUC, this is an internal interface atm, so we could revisit in the future?

Reply:
makes sense, included that in https://github.com/salesforce/mt-dynamo/pull/476